### PR TITLE
Silabs] Update brd2605a rgb led 

### PR DESCRIPTION
**Pull request by bhmanda-silabs**
**Change**
Changes added for RGB LED with wifi_sdk 3.3.3
**Testing**
Validated that lighting-app and lock-app compile and run successfully on BRD4338A and BRD2605A
Commissioning and led(s) are working

### CODE REVIEW

File: examples/platform/silabs/uart.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ extern "C" {
 #include "rsi_board.h"
 #include "rsi_debug.h"
 #include "rsi_rom_egpio.h"
-#include "sl_si91x_usart.h"
+// #include "sl_si91x_usart.h"
```
Comment (rosahay-silabs) [inline examples/platform/silabs/uart.cpp:41]: remove any uncommented lines if not required

File: src/platform/silabs/platformAbstraction/WiseMcuSpam.cpp [hunk 2]
```diff
@@ -36,8 +36,15 @@ extern "C" {
...
 #include "sl_si91x_button_pin_config.h"
+#ifndef SI917_DEVKIT
 #include "sl_si91x_led.h"
 #include "sl_si91x_led_config.h"
+#include "sl_si91x_led_instances.h"
+#else
+#include "sl_si91x_rgb_led.h"
+#include "sl_si91x_rgb_led_config.h"
+#include "sl_si91x_rgb_led_instances.h"
+#endif
```
Comment (rosahay-silabs) [inline src/platform/silabs/platformAbstraction/WiseMcuSpam.cpp:47]:
```suggestion
#endif // SI917_DEVKIT
```

File: src/platform/silabs/platformAbstraction/WiseMcuSpam.cpp [hunk 3]
```diff
@@ -111,7 +119,11 @@ void SilabsPlatform::InitLed(void)
 CHIP_ERROR SilabsPlatform::SetLed(bool state, uint8_t led)
 {
     VerifyOrReturnError(led < SL_LED_COUNT, CHIP_ERROR_INVALID_ARGUMENT);
+#ifndef SI917_DEVKIT
     (state) ? sl_si91x_led_set(ledPinArray[led]) : sl_si91x_led_clear(ledPinArray[led]);
+#else
+    (state) ? sl_si91x_simple_rgb_led_on(SL_RGB_LED_INSTANCE(led)) : sl_si91x_simple_rgb_led_off(SL_RGB_LED_INSTANCE(led));
+#endif
```
Comment (rosahay-silabs) [inline src/platform/silabs/platformAbstraction/WiseMcuSpam.cpp:126]:
```suggestion
#endif // SI917_DEVKIT
```

File: src/platform/silabs/platformAbstraction/WiseMcuSpam.cpp [hunk 4]
```diff
@@ -124,7 +136,11 @@ bool SilabsPlatform::GetLedState(uint8_t led)
 CHIP_ERROR SilabsPlatform::ToggleLed(uint8_t led)
 {
     VerifyOrReturnError(led < SL_LED_COUNT, CHIP_ERROR_INVALID_ARGUMENT);
+#ifndef SI917_DEVKIT
     sl_si91x_led_toggle(ledPinArray[led]);
+#else
+    sl_si91x_simple_rgb_led_toggle(SL_RGB_LED_INSTANCE(led));
+#endif
```
Comment (rosahay-silabs) [inline src/platform/silabs/platformAbstraction/WiseMcuSpam.cpp:143]:
```suggestion
#endif // SI917_DEVKIT
```

File: src/platform/silabs/rs911x/wfx_sl_ble_init.cpp [hunk 5]
```diff
@@ -294,6 +294,7 @@ void SilabsBleWrapper::rsi_ble_add_char_val_att(void * serv_handler, uint16_t ha
 
 uint32_t SilabsBleWrapper::rsi_ble_add_matter_service(void)
 {
+    bleEvent.eventData                                      = new SilabsBleWrapper::sl_wfx_msg_t();
```
Comment (rosahay-silabs) [inline src/platform/silabs/rs911x/wfx_sl_ble_init.cpp:297]: mention this on the PR changes.

File: third_party/silabs/SiWx917_sdk.gni [hunk 6]
```diff
@@ -156,6 +155,19 @@ template("siwx917_sdk") {
       "${wifi_sdk_root}/components/device/silabs/si91x/mcu/drivers/service/power_manager/inc",
     ]
 
+    if (rgb_led) {
+      _include_dirs += [
+        "${wifi_sdk_root}/components/device/silabs/si91x/mcu/drivers/hardware_drivers/rgb_led/inc",
+        "${wifi_sdk_root}/components/device/silabs/si91x/mcu/drivers/hardware_drivers/rgb_led/config",
+      ]
+      defines += [ "SI917_DEVKIT" ]
```
Comment (rosahay-silabs) [inline third_party/silabs/SiWx917_sdk.gni:163]: is this the correct place of this definition? I believe it should be based on the board right and not RGB_LED ?

File: third_party/silabs/silabs_board.gni [hunk 7]
```diff
@@ -56,6 +56,9 @@ declare_args() {
 
   # Temperature Sensor support
   sl_enable_si70xx_sensor = false
+
+  #RGB LED support
+  rgb_led = false
```
Comment (rosahay-silabs) [inline third_party/silabs/silabs_board.gni:61]: `sl_enable_rgb_led`, would be a better name IMO

**Comment by bhmanda-silabs** [comment]
The same changes are merged in https://github.com/project-chip/connectedhomeip/pull/36086 as CI build is failing due to BRD2605A changes are missing.

**Comment by bhmanda-silabs** [comment]
Closing this PR as the changes are merged in https://github.com/project-chip/connectedhomeip/pull/36086 as CI build is failing due to BRD2605A changes are missing.

**Pull request closed without merging**